spinoso-string/src/enc/ascii/mod.rs

```diff
@@ -198,7 +198,15 @@ impl AsciiString {
     pub fn get_char_slice(&self, range: Range<usize>) -> Option<&'_ [u8]> {
         let Range { start, end } = range;
 
-        self.inner.get(start..end).or_else(|| self.inner.get(start..))
+        self.inner.get(start..end).or_else(|| {
+            if start > self.inner.len() {
+                None
+            } else if end <= start {
+                Some(&b""[..])
```

Comment:
This removes a few symbols to make this line read clearer.

```suggestion
                Some(&[])
```

Reply:
Fixed [here](https://github.com/artichoke/artichoke/pull/2102/files#diff-1578dd864d1a9bac39c945d985837664c4c4655d41f16fdba68ea16ecef49c4aR205)